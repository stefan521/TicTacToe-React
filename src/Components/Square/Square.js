import React from 'react';

export default function Square(props) {
  return (
    <button className="square" onClick={props.onClick}
      style={{background: props.color}}>
      {props.value}
    </button>
  );
}
