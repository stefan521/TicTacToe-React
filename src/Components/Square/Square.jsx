// @flow
import React from 'react';

type Props = {
  onClick: Function,
  winSquare: boolean,
  value: string
};

export default function Square(props: Props) {
  const { onClick } = props;
  const { winSquare } = props;
  const { value } = props;

  return (
    <div className="grid-item">
      <button
        type="button"
        className={winSquare ? 'win-square square' : 'square'}
        onClick={onClick}
      >
        {value}
      </button>
    </div>
  );
}
