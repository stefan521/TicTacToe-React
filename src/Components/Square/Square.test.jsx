import React from 'react';
import { shallow } from 'enzyme';
import Square from './Square.jsx';

describe('Square', () => {
  const clickHandler = jest.fn();
  const value = 'X';

  it('renders with text and background color', () => {
    const wrapper = shallow(<Square
      winSquare={false}
      onClick={clickHandler}
      value={value}
    />);

    expect(wrapper.props().className).toEqual('grid-item');
    expect(wrapper).toMatchSnapshot();
  });
});
