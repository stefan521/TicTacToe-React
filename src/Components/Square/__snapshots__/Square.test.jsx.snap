// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square renders with text and background color 1`] = `
<div
  className="grid-item"
>
  <button
    className="square"
    onClick={[MockFunction]}
    type="button"
  >
    X
  </button>
</div>
`;
